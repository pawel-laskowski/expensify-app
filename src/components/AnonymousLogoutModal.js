import React from 'react'
import Modal from 'react-modal'

export const AnonymousLogoutModal = (props) => {
  return (
    <Modal
      isOpen={props.modalIsOpen}
      onRequestClose={props.closeModal}
      ariaHideApp={false}
      contentLabel="RemoveExpense"
      closeTimeoutMS={200}
      className="modal modal--logout"
    >
      <h3 className="modal__titel">If you logout all expenses will be removed. Login with Google to save your expenses.</h3>
      <div className="modal__buttons">
        <button className="button button--secondary" onClick={props.startLogout}>Logout</button>
        <button className="button button" onClick={props.convertAccountToPermanent}>Login with Google</button>    
      </div>
    </Modal>
  )
} 

export default AnonymousLogoutModal