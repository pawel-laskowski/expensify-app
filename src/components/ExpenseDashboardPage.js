import React from 'react'
import ExpenseList from './ExpenseList'
import ExpenseListFilters from './ExpenseListFIlters'

const ExpenseDashboardPage = () => {
  return (
    <div>
      <ExpenseListFilters />
      <ExpenseList />
    </div>
  )
}

export default ExpenseDashboardPage