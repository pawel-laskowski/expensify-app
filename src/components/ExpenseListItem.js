import React from "react";
import { Link } from "react-router-dom";

const ExpenseListItem = ( { description, amount, id, createdAt}) => (
  <div>
    <Link to={`edit/${id}`}>
      <h2>Description: {description}</h2>
    </Link>
    <p>amount: {amount}</p>  
  </div>
)

export default ExpenseListItem