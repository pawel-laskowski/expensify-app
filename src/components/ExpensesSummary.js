import React from "react";
import { connect } from 'react-redux'
import { Link } from "react-router-dom";
import numeral from "numeral";
import selectExpensesTotal from "../selectors/expenses-total";
import selectExpenses from "../selectors/expenses"

export const ExpensesSummary = (props) => {
  const expensesCount = props.expenses.length
  const expensesTotal = selectExpensesTotal(props.expenses)
  const expensesWord = expensesCount === 1 ? 'expense' : 'expenses'
  const formattedExpensesTotal = numeral(expensesTotal / 100).format('$0,0.00')
  const hiddenExpenses = props.allExpenses.length - props.expenses.length
  return (
    <div className="page-header">
      <div className="content-container">
        <h1 className="page-header__titel">Viewing <span>{expensesCount}</span> {expensesWord} totalling <span>{formattedExpensesTotal}</span></h1>
        {hiddenExpenses > 0 && <p>{hiddenExpenses} hidden expenses because of filters. Remove filters to see all expenses.</p>}
        <div className="page-header__actions">
          <Link className="button" to="/create">Add Expense</Link>
        </div>
      </div>
    </div>
  )
}

const mapStateToProps = (state, props) => {
  return {
    expenses: selectExpenses(state.expenses, state.filters),
    allExpenses: state.expenses
  }
}

export default connect(mapStateToProps)(ExpensesSummary)
