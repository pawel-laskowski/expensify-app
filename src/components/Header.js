import React from 'react'
import { NavLink } from 'react-router-dom'

const Header = () => (
  <header>
    <h1>Expensify</h1>
    <NavLink to="/">Dashboard</NavLink>
    <NavLink to="/create">Create Expense</NavLink>
  </header>
)

export default Header