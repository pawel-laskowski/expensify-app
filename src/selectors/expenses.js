import moment from "moment"

//get VisibleExpenses
export default (expenses, { text, sortBy, startDate, endDate }) => {
  return expenses.filter((expense) => {
    const createdAtMoment = moment(expense.createdAt)
    const startDateMatch = startDate ? startDate.isSameOrBefore(createdAtMoment, 'day'): true
    const endDateMatch = endDate ? endDate.isSameOrAfter(createdAtMoment, 'day'): true
    const textMatch = typeof text !== 'string' || expense.description.toLocaleLowerCase().includes(text.toLocaleLowerCase())
    
    return startDateMatch && endDateMatch && textMatch
  }).sort((a, b) => {
    if (sortBy === 'date') {
      return a.createdAt < b.createdAt ? 1 : -1
    } else if (sortBy === 'amount') {
      return a.amount < b.amount ? 1 : -1
    }
  })
}