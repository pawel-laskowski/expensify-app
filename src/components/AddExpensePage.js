import React from 'react'
import { connect } from "react-redux";
import { useNavigate  } from 'react-router-dom';
import { addExpense } from '../actions/expenses'
import ExpenseForm from './ExpenseForm'

export const AddExpensePage = (props) => {
  let navigate = useNavigate();
  const onSubmit = (expense) => {
    props.addExpense(expense)
    navigate('/')
  }

  return (
    <div>
      <h1>Add Expense </h1>
      <ExpenseForm 
        onSubmit={onSubmit}
      />
    </div>
  )
}

const mapDispatchToProps = (dispatch) => ({
  addExpense: (expense) => dispatch(addExpense(expense))
})

export default connect(undefined, mapDispatchToProps)(AddExpensePage)