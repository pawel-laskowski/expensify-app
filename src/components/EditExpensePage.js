import React from 'react'
import { connect } from "react-redux";
import { useParams, useNavigate } from 'react-router-dom';
import { startEditExpense, startRemoveExpense } from '../actions/expenses';
import ExpenseForm from './ExpenseForm';
import RemoveExpenseModal from './RemoveExpenseModal';

export const EditExpensePage = (props) => {
  const navigate = useNavigate()
  const { id } = useParams()
  const expense = props.expenses.find(expense => expense.id === id)
  const [modalIsOpen, setIsOpen] = React.useState(false)

  const onSubmit = (expense) => {
    props.startEditExpense(id, expense)
    navigate('/')
  }
  const onRemove = () => {
    props.startRemoveExpense({ id })
    navigate('/')
  }
  const openModal = () => {
    setIsOpen(true);
  }
  const closeModal = () => {
    setIsOpen(false);
  }

  return (
    <div>
      <div className="page-header">
        <div className="content-container">
          <h1 className="page-header__title">Edit Expense</h1>
        </div>
      </div>
      <div className="content-container">
        <ExpenseForm 
          expense={expense}
          onSubmit={onSubmit}
        />
        <button className="button button--secondary" onClick={openModal}>Remove Expense</button>
        <RemoveExpenseModal 
          modalIsOpen={modalIsOpen}
          closeModal={closeModal}
          onRemove={onRemove}
        />        
      </div>
    </div>
  )
}

const mapStateToProps = (state, props) => {
  return {
    expenses: state.expenses
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
    startRemoveExpense: ({ id }) => dispatch(startRemoveExpense({ id }))
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage)