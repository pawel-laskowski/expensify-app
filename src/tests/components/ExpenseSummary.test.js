import React from 'react'
import { shallow } from "enzyme"
import { ExpensesSummary } from "../../components/ExpensesSummary"
import expenses from "../fixtures/expenses"

test('should correctly render ExpensesSummary with 1 expense', () => {
  const wrapper = shallow(<ExpensesSummary expenses={[expenses[0]]} allExpenses={expenses}/>)
  expect(wrapper).toMatchSnapshot()
})

test('should correctly render ExpensesSummary with multiple expenses', () => {
  const wrapper = shallow(<ExpensesSummary expenses={expenses} allExpenses={expenses}/>)
  expect(wrapper).toMatchSnapshot()
})