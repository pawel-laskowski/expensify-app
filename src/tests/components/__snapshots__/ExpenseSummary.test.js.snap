// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpensesSummary with 1 expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__titel"
    >
      Viewing 
      <span>
        1
      </span>
       
      expense
       totalling 
      <span>
        $1.95
      </span>
    </h1>
    <p>
      2
       hidden expenses because of filters. Remove filters to see all expenses.
    </p>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should correctly render ExpensesSummary with multiple expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__titel"
    >
      Viewing 
      <span>
        3
      </span>
       
      expenses
       totalling 
      <span>
        $1,141.95
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
